lib/dialects/mysql/data-types.js

```diff
@@ -43,13 +43,7 @@ module.exports = BaseTypes => {
   inherits(BLOB, BaseTypes.BLOB);
 
   BLOB.parse = function(value, options, next) {
-    const data = next();
```

Comment:
I dont think so this works, can you show me a failing case

Reply:
I'm not sure where the right place to add test case for this.

What I did here is only removed
https://github.com/sequelize/sequelize/blob/master/lib/dialects/mysql/data-types.js#L48-L50
what causes returning null instead of empty buffer.

Reply:
Ok show me example of what you are trying to solve, paste it here

Reply:
Something like:
expect((await model.findOne({ some_where_clause })).empty_buffer_column).to.equal(Buffer.from(""));

In fact `mysql2` what is used returns empty buffer in select queries, when `sequelize` catches that and returns null instead.